hamcrest-core/src/main/java/org/hamcrest/MatcherAssert.java

```diff
@@ -1,16 +1,19 @@
 package org.hamcrest;
 
-
 public class MatcherAssert {
     public static <T> void assertThat(T actual, Matcher<? super T> matcher) {
         assertThat("", actual, matcher);
     }
     
     public static <T> void assertThat(String reason, T actual, Matcher<? super T> matcher) {
+    	assertThat(selfDescribingString(reason), actual, matcher);
+    }
+    
+    public static <T> void assertThat(SelfDescribing reason, T actual, Matcher<? super T> matcher) {
```

Comment:
I would suggest adding some Javadoc that shows how this would be used with Lambdas.

I'm surprised that there isn't any Javadocs in this class at all :-(